### DISCUSSION TITLE
Unhide menu bar?

### DISCUSSION THREAD

**Discussion by Nietzscheanman** [keepassxreboot/keepassxc#10706] [Q&A] [answered]
Hello,

I was messing around with the options and went to view -> hide menu, and now I cannot seem to find how to unhide it?

As far as I can tell the "user interface" options in "application settings" do not include such an option, and there seems to be nowhere to right-click it back on etc.?

Any help appreciated - thanks!

**Comment by phoerious** [comment, marked as answer]
Alt should temporarily show the menu bar again.

**Reply by Nietzscheanman** [reply]
Solved, thanks! :)

**Reply by noraj** [reply]
Alt does nothing on ArchLinux/KDE/Wayland

**Comment by d1nuc0m** [comment]
My two cents in case someone else finds this - somehow on Linux pressing alt didn't show the menubar again. Closed KeepassXC, opened ~/.config/keepassxc/keepassxc.ini and set [GUI]/HideMenubar=false (instead of true). Reopened and it works

**Reply by zkkv** [reply]
For me it was under `~/.var/app/org.keepassxc.KeePassXC/config/keepassxc`

**Reply by noraj** [reply]
Setting `HideMenubar=false` in `~/.config/keepassxc/keepassxc.ini` didn't work.

Note: `~/.var/app/org.keepassxc.KeePassXC/config/keepassxc` is the path for Flatpak installation.

**Comment by EliteAMDGamer** [comment]
The issue is still there, on arch Garuda KDE, I do use the global menu. On version `2.7.9-2` alt dosen't work. or adding `[GUI]/HideMenubar=false` to the ini it just clears it on opening the app. I'm trying to export my passwords so this is kinda critical. Maybe a hot key like `ctrl + alt + e` or something. 
The wayland trick of clicking off the window an clicking back on the app is a no go as well. 

Name            : keepassxc
Version         : 2.7.9-2

```Operating System: Garuda Linux 
KDE Plasma Version
KDE Frameworks Version: 6.4.0
Qt Version: 6.7.2
Kernel Version: 6.10.0-1-cachyos (64-bit)
Graphics Platform: Wayland```

**Reply by droidmonkey** [reply]
You really just need to delete the existing line item that is `HideMenubar=true`.

You can also go to application settings, scroll to the bottom, and click "Reset settings"

**Reply by EliteAMDGamer** [reply]
No luck, even with a reset an full remove an reinstall.  Also tried git.  
I tried adding it to the top then the bottom prior to reset and after. 

~Update Removing the global menu seems to allow it to open in the app as it should. 

```
[GUI]
ApplicationTheme=classic
ColorPasswords=true
CompactMode=false
MinimizeOnClose=true
MinimizeOnStartup=true
MinimizeToTray=true
MonospaceNotes=true
ShowTrayIcon=true
ToolButtonStyle=0
TrayIconAppearance=monochrome-light
HideMenubar=false
```

**Reply by droidmonkey** [reply]
> Removing the global menu seems to allow it to open in the app as it should.

Then your problem is with KDE for not showing the global menu properly?

**Reply by EliteAMDGamer** [reply]
Yeah, thank you. I'll file a bug report with kde. as far as i know its only this app but i'll check some others.

**Reply by materemias** [reply]
@EliteAMDGamer can you reference the KDE bug report? I have the same issue and would like to follow up

**Reply by materemias** [reply]
I am on Arch KDE wayland, and I cannot invoke the application menu of KeepassXC. I was using global menu but removed it for now, I checked the ini file, entered HideMenubar=false manually, nothing gets to it (Alt does nothing as well).  Do you have any other idea @droidmonkey ?

**Reply by EliteAMDGamer** [reply]
> @EliteAMDGamer can you reference the KDE bug report? I have the same issue and would like to follow up

Sure, here it is. 
https://bugs.kde.org/show_bug.cgi?id=490430

**Reply by alexisju** [reply]
Same issue and same config (Arch/KDE/Wayland). The workaround is to open KeePassXC in X11, like this :

`QT_QPA_PLATFORM=xcb keepassxc`

**Reply by noraj** [reply]
Even opening the in X11 doesn't work. The only workaround for me was to remove the global menu from the task bar.

**Reply by droidmonkey** [reply]
There is a bug in the global menu, not us

**Reply by joaociocca** [reply]
> Same issue and same config (Arch/KDE/Wayland). The workaround is to open KeePassXC in X11, like this :
> 
> `QT_QPA_PLATFORM=xcb keepassxc`

Running Garuda+KDE here, this workaround helped and the menu showed up on the global menu task bar.

Too bad it then shows up bright white killing my eyes :rofl: (ohhh! there's the menu for theme, now I can put it back in dark mode)

**Comment by crossan007** [comment]
Running KeePassXC 2.7.9 on Kubuntu 24.10 still has this bug.

I was able to restore the menu bar by _removing_ the line `HideMenubar=true` from `~/.config/keepassxc/keepassxc.ini`

**Comment by dougg0k** [comment]
KDE6

Same issue on Wayland and X11, QT workaround does not work. Nor adding `HideMenuBar=false`.

---

I actually found something that worked.

In KDE
1. Open Background Services 
2. Stop and (Uncheck) Application menus daemon 
3. Done

**Reply by PatRiot1337** [reply]
> KDE6
> 
> Same issue on Wayland and X11, QT workaround does not work. Nor adding `HideMenuBar=false`.
> 
> I actually found something that worked.
> 
> In KDE
> 
>     1. Open Background Services
> 
>     2. Stop and (Uncheck) Application menus daemon
> 
>     3. Done

THIS WORKED! Was the only solution. Tried all other hints, but they did not work.

Same issue here on Garuda, KDE6, your solution was the only one that worked. Keepassxc sometimes was in light theme, sometimes black mode, my eyes..., so this helped me to set it to dark (same as [GUI]/ApplicationTheme=dark), then restarted application menus daemon again, maybe it should be on? (Not for Keepassxc but for the system itself?).

Thanks man, @dougg0k ! 🙏🏻 Exactly one year later! :D

**Reply by cyon-e** [reply]
same issue here on KDE via Alpine... thanks for the hint
that a Qt app is not supporting KDE correctly baffles me

**Comment by shotor** [comment]
I got the global menu working by changing: `qt.platformTheme = "qt5ct";` to `kde` in my `nix` config (removing it altogether works too).

If I set it to `qt5ct` it installs:

- `libsForQt5.qt5ct`
- `qt6Packages.qt6ct`

Which is just `qt5ct` and `qt6ct` on `archlinux`

Found this bug report and can reproduce: https://bugs.kde.org/show_bug.cgi?id=483335

```
journalctl --follow | grep org.kde.plasma.appmenu

# start keepassxc

kded6[2025]: org.kde.plasma.appmenu: Got an error
```

Would have been nice to get more info from this error, not sure if possible.

Issue mentions installing `plasma5-integration`. Changing to `kde` in my case installs `libsForQt5.plasma-integration`, which I assume is the same

**Reply by noraj** [reply]
It may not be all, on ArchLinxu I have [plasma-integration](https://archlinux.org/packages/extra/x86_64/plasma-integration/) installed and forcing `QT_QPA_PLATFORMTHEME=kde keepassxc` doesn't make KeePassXC use the KDE global menu.

**Reply by noraj** [reply]
However, I can see the error you mentioned:

```
➜ journalctl --follow | grep org.kde.plasma.appmenu
mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
```

**Reply by noraj** [reply]
> It may not be all, on ArchLinxu I have [plasma-integration](https://archlinux.org/packages/extra/x86_64/plasma-integration/) installed and forcing `QT_QPA_PLATFORMTHEME=kde keepassxc` doesn't make KeePassXC use the KDE global menu.

Oh sorry, KeepassXC is still using Qt5, so installing [plasma5-integration](https://archlinux.org/packages/extra/x86_64/plasma5-integration/) just worked even without me forcing the `QT_QPA_PLATFORMTHEME` value.

**Reply by noraj** [reply]
> However, I can see the error you mentioned:
> 
> ```
> ➜ journalctl --follow | grep org.kde.plasma.appmenu
> mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
> mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
> mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
> mars 04 21:29:29 archpowa kded6[1229]: org.kde.plasma.appmenu: Got an error
> ```

With [plasma5-integration](https://archlinux.org/packages/extra/x86_64/plasma5-integration/) installed and the global menu working for KeepassXC, the error still triggers.

**Reply by noraj** [reply]
I documented this here: https://wiki.archlinux.org/title/Qt#KDE_Global_Menu_not_working

**Comment by KitchM** [comment]
I have no other program, nor have I ever heard of a program, where the menu bar is hidden by default and comes back by pressing the Alt key.  Worse yet, there is a failure to use the right-click over the button and/or title bar to find a context menu whereby the bars are configured.  All I get is a four-way arrow.  This is an odd programming error.  Since others do not do this, it is clearly a programming error IMHO.  Proper programming requires not only intuitive design, but also consistency with expectations in normal GUI usage.  This is lacking.

**Reply by droidmonkey** [reply]
Menubar is NOT hidden by default.

Alt key is standard for showing or focusing on the menu bar.

**Comment by KitchM** [comment]
Just received notification of new post here, but it turned out that there wasn't a new post.  Very confusing.

Anyway, as a clarification to the posts here, the Alt key is a key to be used in combination with other keys to change, or alternate, the function of these other keys.  This is the foundational standard of the key.

For instance, a standard usage is to press Alt-F to pull down and display the File menu.  According to Wikipedia, the Alt key by itself is not known to simple unhide a menu bar.  I would like to know more about any such standard in the industry.

**Reply by joaociocca** [reply]
The notification was probably caused by [my comment](https://github.com/keepassxreboot/keepassxc/discussions/10706#discussioncomment-13063319). Back to your subject... from my Windows days, I remember that pressing Alt itself alone would show hidden menu bars and allow me to navigate thru them either by using arrows or shortcut letters.
Not sure if this behavior repeated in Linux's DEs.

**Comment by KitchM** [comment]
Oh, I get it.  Thanks.

I did try a number of random programs and none reacted to the Alt key by itself, except for Thunderbird, which only highlighted the File menu.

**Comment by meatspinner** [comment]
For people on Garuda, the only solution I found was using the Flatpak package. Its not that big of a deal but its a shame.

**Reply by joaociocca** [reply]
you probably missed my comment up there https://github.com/keepassxreboot/keepassxc/discussions/10706#discussioncomment-13063319

<img width="1886" height="935" alt="image" src="https://github.com/user-attachments/assets/6a38be94-32fe-4a2a-853e-e69b84f55509" />

**Reply by meatspinner** [reply]
Yeah, sorry, I shouldn't have said only. I didn't miss it but to me personally its a bit inconvenient to do that if you prefer to use Wayland.

**Reply by joaociocca** [reply]
I'm feeling a bit dumb at the moment, if you'd mind could you explain what the problem would be - besides being a workaround, instead of a true fix from keepassxc itself which I agree would be best. I ask this because I'm using Garuda/KDE/Wayland, and except for adding that snippet, I see no difference in keepass' behavior.

**Reply by meatspinner** [reply]
My bad, I read that fast the first time and I thought I would have to restart the session as X11, didn't realize I could just open the app on the terminal or edit the .desktop file.

**Comment by gwholland3** [comment]
Apologies if this was already discussed somewhere on this issue, but I am running into the same problem on macOS KeePassXC. At some point (I'm not sure when), I must have hidden the menubar, and now I cannot get it to come back - checking "Show menubar" in settings doesn't work, neither does manually setting `HideMenubar` to `false` in `keepassxc.ini`.

Is there a resolution to this on macOS? I'm aware that I can get the menubar dropdowns to show up when "Show menubar" is unselected by pressing the option key, but that is not convenient - even then, you only see the dropdown options, not the button the dropdown is descending from.

For what it's worth, this problem also appears to have been brought up in #13060 for KDE.

**Reply by droidmonkey** [reply]
Are you using Display Link? If so, then you need to enable "Allow Screen Capture". https://keepassxc.org/docs/KeePassXC_UserGuide#_screenshot_security

**Reply by gwholland3** [reply]
I'm not sure I follow... My problem is that only the menubar is hidden - the rest of the application appears fine.

I expect to see something like this (from a sample internet picture):
<img width="1534" height="406" alt="Captura de pantalla 2026-03-09 a la(s) 12 43 30" src="https://github.com/user-attachments/assets/463c2f81-86f5-4170-baa2-71b37e9b9a66" />

Instead, I see something like this:
<img width="782" height="198" alt="Captura de pantalla 2026-03-09 a la(s) 12 45 22" src="https://github.com/user-attachments/assets/cf378405-0aab-41e4-828a-c969b69484d5" />

I have tried selecting "Allow Screen Capture", but that did not help. As for your question about Display Link - I do have the [`displaylink`](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/d/displaylink.rb) app installed through Homebrew, but I rarely use it and experience the problem of the hidden menubar even when not connected to any external display devices.

**Reply by droidmonkey** [reply]
On macos, application menubars are placed at the very top of the screen.

**Reply by gwholland3** [reply]
Ah I see now, thank you! I'll just leave "Show menubar" selected in my settings then, since disabling it makes it so phantom menubar dropdowns appear when I press the option key.

**Reply by droidmonkey** [reply]
That is interesting because we dont allow hiding the menubar on macos (i checked). Are you running a hackintosh or something?

**Reply by gwholland3** [reply]
No, just standard macOS Sequioa on a 2019 MacBook Pro. I have this option in my settings:
<img width="514" height="320" alt="Captura de pantalla 2026-03-12 a la(s) 23 06 48" src="https://github.com/user-attachments/assets/833799fc-20ac-4374-82e0-6c8b59a9ec70" />

And if it's deselected, pressing the option key yields this behavior - a dropdown menu from nowhere:
<img width="526" height="714" alt="Captura de pantalla 2026-03-12 a la(s) 23 07 24" src="https://github.com/user-attachments/assets/4583f5fc-c541-4fe9-85d5-5eeef1c4d50f" />

**Reply by droidmonkey** [reply]
Oh wow, ok we need to hide that option (it is hidden in the view menu) and actively ignore it being set.
